src/theory/strings/regexp_solver.cpp

```diff
@@ -174,14 +79,28 @@ void RegExpSolver::check()
       << std::endl;
   if (!addedLemma)
   {
+    // get all memberships
+    std::vector<Node> allMems;
+    for (const std::pair<const Node, std::vector<Node> >& mr : mems)
+    {
+      for (const Node& m : mr.second)
+      {
+        bool polarity = m.getKind() != NOT;
+        if (polarity || !options::stringIgnNegMembership())
+        {
+          allMems.push_back(m);
+        }
+      }
+    }
```

Comment:
Just a thought: I am not quite sure if it is better to compute this once and then use it twice or to just compute it twice below (saving a bunch of copies and allocations).

Reply:
I think I may refactor this on a future PR; I don't think this is a big deal performance wise.